hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java

```diff
@@ -172,14 +173,47 @@ public TransactionMetadata preHandleUpdateTokenFeeSchedule(
     public TransactionMetadata preHandlePauseToken(
             @NonNull final TransactionBody txn, @NonNull final AccountID payer) {
         Objects.requireNonNull(txn);
-        throw new NotImplementedException();
+        final var op = txn.getTokenPause();
+        final var meta =
+                new SigTransactionMetadataBuilder(accountStore).payerKeyFor(payer).txnBody(txn);
```

Comment:
We might pass in `SigTransactionMetadataBuilder` into the methods in future changes.